test/python/bfloat16/test_conv2dbackpropfilter_nchw.py

```diff
@@ -0,0 +1,116 @@
+# ==============================================================================
+#  Copyright 2019 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# ==============================================================================
+"""nGraph TensorFlow bridge Conv2d operation test
+
+"""
+import pytest
+
+import tensorflow as tf
+import numpy as np
+import os
+from tensorflow.python.ops import nn_ops
+import ngraph_bridge
+
+#Tests Ngraph Op: ConvolutionBackpropFilters with data format NCHW
+#TF Op: conv2d_backprop_filter
+
+np.random.seed(5)
+#Inputs
+N = 1
+H = 7
+W = 6
+C = 2
+
+I = C
+O = 2
+FW = 3
+FH = 3
+
+input_sizes_nchw = [N, C, H, W]
+input_sizes_nhwc = [N, H, W, C]
+filter_size_hwio = [FH, FW, I, O]
+out_backprop_valid = [1, 2, 3, 2]
+out_backprop_same = [1, 2, 4, 3]
+out_backprop_in_sizes = {"VALID": out_backprop_valid, "SAME": out_backprop_same}
+stride_nhwc = [1, 2, 2, 1]
+stride_nchw = [1, 1, 2, 2]
+
+
+#TF graph
+def tf_model(padding):
+    t1 = tf.placeholder(dtype=tf.float32, shape=input_sizes_nhwc, name='t1')
+    t2 = tf.constant(filter_size_hwio, dtype=tf.int32, name='t1')
+    t3 = tf.placeholder(
+        dtype=tf.float32, shape=out_backprop_in_sizes[padding], name='t3')
+
+    #reshaping the out_backprop to NHWC since TF does not support NCHW
+    t3 = tf.transpose(t3, [0, 2, 3, 1])
+
+    #Cast dtype to bfloat16 for TF because NNP casts ng_model inputs
+    t1 = tf.cast(t1, dtype=tf.bfloat16)
+    t3 = tf.cast(t3, dtype=tf.bfloat16)
+
+    filt = nn_ops.conv2d_backprop_filter(
+        t1, t2, t3, stride_nhwc, padding=padding, data_format='NHWC')
+
+    #Cast dtype back to float32 similar to NNP
+    filt = tf.cast(filt, dtype=tf.float32)
+    return filt, t1, t3
+
+
+#Ngraph Graph
+def ng_model(padding):
+    t1 = tf.placeholder(dtype=tf.float32, shape=input_sizes_nchw, name='t1')
+    t2 = tf.constant(filter_size_hwio, dtype=tf.int32, name='t1')
+    t3 = tf.placeholder(
+        dtype=tf.float32, shape=out_backprop_in_sizes[padding], name='t3')
+
+    filt = nn_ops.conv2d_backprop_filter(
```

Comment:
do we need to transpose the output of either tf or ngraph before comparing?

Reply:
No, output is same shape as filter which is a constant. the shapes are same for both TF and Ngraph graph